hudi-common/src/main/java/org/apache/hudi/common/BloomFilterFactory.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hudi.common;
+
+import org.apache.hadoop.util.hash.Hash;
+
+/**
+ * A Factory class to generate different versions of {@link BloomFilter}
+ */
+public class BloomFilterFactory {
+
+  /**
+   * Creates a new {@link BloomFilter} with the given args
+   * @param numEntries total number of entries
+   * @param errorRate max allowed error rate
+   * @param bloomFilterVersion bloom filter version
+   * @return the {@link BloomFilter} thus created
+   */
+  public static BloomFilter createBloomFilter(int numEntries, double errorRate, int bloomFilterVersion) {
+        return createBloomFilter(numEntries, errorRate, Hash.MURMUR_HASH, bloomFilterVersion);
+  }
+
+  /**
+   * Creates a new {@link BloomFilter} with the given args
+   * @param numEntries total number of entries
+   * @param errorRate max allowed error rate
+   * @param hashType type of the hashing function (see
+   * {@link org.apache.hadoop.util.hash.Hash}).
+   * @param bloomFilterVersion bloom filter version
+   * @return the {@link BloomFilter} thus created
+   */
+  public static BloomFilter createBloomFilter(int numEntries, double errorRate, int hashType, int bloomFilterVersion) {
+    if(bloomFilterVersion == SimpleBloomFilter.VERSION){
```

Comment:
I wonder if we are mixing concepts of versioning the bytes written by the bloom filter with the type of BloomFilter? 

how would we handle a different serialization for the dynamic bloom filter itself for e.g? Would that need to be a separate `DynamicBloomFilterV2` implementation? if so, it makes sense to call this `typeCode` than `VERSION`

Reply:
Then, don't you think that logic needs to be contained within the BloomFilter impl. why the metadata footer needs to know of diff ser versions within the same BloomFilter impl. I introduced versions to make BloomFilter implementations iterative. Tomorrow if we come up with a new DynamicBloomFilterV2, we will give it version no 2(simple -> 0, DynamicBloomFilter -> 1). If within the same BloomFilterImpl, if ser format is going to change, I feel its the responsibility of the impl class to ser that info and deser using the same and not expose that to the footer. 
Let me know what you think.

Reply:
No. i think we are on the same page. just saying instead of calling it versions, lets just call it type code. `TYPE_CODE` .. What we have is different types of bloom filters.

Reply:
ok, I can call it TYPE_CODE and add that to parquet metadata. 
So, are there chances to introduce different versions of DynamicBloomfilter? If yes, we might have to add a version within DynamicBloomFilter and encode the same during serialization(which the metadata is not aware of). Is that what you were suggesting?

Reply:
yes I was wondering if we need a version on top of type code? but dont think we would version this often per se.. So may be just have type code now is ok.